### PR TITLE
lib/modules: Change mkAliasOptionModule to use the priority for the alias

### PR DESCRIPTION
**Pull request by cdepillabout**
This commit changes the `mkAliasOptionModule` function to make sure that the priority for the aliased option is propagated to the non-aliased option.

This also affects the `mkRenamedOptionModule` function in a similar fashion.

This also removes the `mkAliasOptionModuleWithPriority` function, since its functionality is now subsumed by `mkAliasOptionModule`.

This change was recommended by @nbp to see what would be the effect of doing this:
https://github.com/NixOS/nixpkgs/pull/53397#discussion_r245487432

###### Motivation for this change

The motivation for this change (as well as a use-case) is listed in https://github.com/NixOS/nixpkgs/pull/53397#issue-242220506.

Pinging @nbp, since he suggested this change.  Also pinging @Infinisil since he helped with the original PR that inspired this change.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/tests/modules/alias-with-priority-can-override.nix [hunk 1]
```diff
@@ -32,7 +32,7 @@ with lib;
 
   imports = [
     # Create an alias for the "enable" option.
-    (mkAliasOptionModuleWithPriority [ "enableAlias" ] [ "enable" ])
+    (mkAliasOptionModule [ "enableAlias" ] [ "enable" ])
 
     # Disable the aliased option, but with a default (low) priority so it
```
Comment (danbst) [inline lib/tests/modules/alias-with-priority-can-override.nix]: the comment says `with a default (low) priority`, but actual code is `enableAlias = lib.mkForce false;`

Comment (cdepillabout) [reply lib/tests/modules/alias-with-priority-can-override.nix]: @danbst Good catch!

I fixed this in commit dcbd136.

**Comment by cdepillabout** [comment]
@nbp Would you be able to take a look at this?  It might be nice to get this in for 19.03.

**Comment by oxij** [comment]
I just discovered this stuff in the lib, noticed it was introduced in #53397, but I don't understand the use case for `withPriority = false`.

Having read the aforementioned thread I still don't understand what is going on here, wasn't it just a bug that `mkAliasOptionModule` didn't copy the priority? Shouldn't `mkRenamed*` do the same too?

I.e. why that second function was introduced in the first place?

Shouldn't this PR just remove all `withPriority` stuff and always propagate the priority?

**Comment by cdepillabout** [comment]
@oxij Thanks for doing a review of this.  You bring up some good questions.  Let me try to answer them.

> Having read the aforementioned thread I still don't understand what is going on here, wasn't it just a bug that `mkAliasOptionModule` didn't copy the priority?

Originally, the documentation for the functions in question mentioned that they don't copy the priority.  I assumed it wasn't a bug, but intentional.  In the few previous PRs I sent surrounding this issue, no one seemed to think it was a bug.

This PR changes `mkAliasOptionModule` (and `mkRenamedOptionModule`) to copy over the priority.

My assumption is that this isn't really "fixing a bug", but adding an additional feature to these functions.

I think that most people who use `mkAliasOptionModule` (and `mkRenamedOptionModule`) will want this new functionality.

> Shouldn't `mkRenamed*` do the same too?

Yes, this PR changes `mkRenamedOptionModule` to also copy over the priority.

> Why was that second function (`mkAliasOptionModuleWithPriority`) introduced in the first place?

`mkAliasOptionModuleWithPriority` was introduced to do the same thing as `mkAliasOptionModule`, but also additionally copy over the priority.

It makes sense to have `mkAliasOptionModuleWithPriority` if you think that `mkAliasOptionModule` shouldn't copy over the priority.  Since this PR changes `mkAliasOptionModule` to copy over the priority, then `mkAliasOptionModuleWithPriority` is no longer needed.

> Shouldn't this PR just remove all `withPriority` stuff and always propagate the priority?

Yeah, that is one option.  I'd be fine with that (since it would make the code slightly simpler), but I'm not sure why `mkAliasOptionModule` originally did not copy over the priority.  If this was just an oversight, then I imagine it wouldn't have been added to the documentation.  If it was intentional, then I didn't want to completely change it if someone was relying on that functionality.

This PR *does* change that functionality, but at least it provides a backdoor for someone to use who is depending on that functionality (by leaving the option available in `doRename`).

**Comment by oxij** [comment]
I see. I guess it would be nice to hear what @nbp thinks about this then. I see no explanation for not propagating the priority in the first place.

**Comment by Infinisil** [comment]
I think this is fine, this change makes sense, and I also even doubt anybody relied on the previous behavior

**Pull request merged**